public/app/features/datasources/utils.ts

```diff
@@ -45,3 +47,10 @@ function incrementLastDigit(digit: number) {
 function getNewName(name: string) {
   return name.slice(0, name.length - 1);
 }
+
+export const constructDataSourceExploreUrl = (dataSource: DataSourceSettings<DataSourceJsonData, {}>) => {
+  const exploreState = JSON.stringify({ datasource: dataSource.name, context: 'explore' });
+  const exploreUrl = urlUtil.renderUrl('/explore', { left: exploreState });
```

Comment:
Should this function make use of `config.appSubUrl` for when users run grafana on a subpath (e.g. `localhost:3000/grafana`)?

Reply:
Done